### PR TITLE
Export and import rack from __str__.

### PR DESCRIPTION
**Pull request by fossilet**
One feature request from our team is that they want to import from human readable representation of racks in CSV file rather than filing them with int ID's. After I implemented this, I find this does not break the models of ralph, since **str** of rack is unique for each one.

### CODE REVIEW

File: src/ralph/data_importer/resources.py [hunk 1]
```diff
@@ -233,7 +234,7 @@ class DataCenterAssetResource(RalphModelResource):
     rack = fields.Field(
         column_name='rack',
         attribute='rack',
-        widget=ImportedForeignKeyWidget(physical.Rack),
+        widget=RackWidget(physical.Rack),
```
Comment (szok) [inline src/ralph/data_importer/resources.py:237]: Does it work with CLI? with the flag MAP_IMPORTED_ID_TO_NEW_ID ? (import from old ralph)

File: src/ralph/data_importer/widgets.py [hunk 2]
```diff
@@ -217,3 +217,29 @@ def render(self, value):
...
+            return None
+        else:
+            try:
+                racks = Rack.objects.all()
+                data = ({'id': rack.id, '__str__': str(rack)} for rack in racks)
+                for datum in data:
+                    if datum['__str__'] == value:
+                        value = Rack.objects.get(id=datum['id'])
+                        break
+            except (ValueError, ServiceEnvironment.DoesNotExist):
```
Comment (szok) [inline src/ralph/data_importer/widgets.py]: Rack.DoesNotExists

File: src/ralph/data_importer/resources.py (hunk 1 above)
Comment (fossilet) [inline src/ralph/data_importer/resources.py:237]: No, it prints `[08:52:53] WARNING Record with pk 1 not found for model assetmodel of 'model'` and `Some of records were not imported`. It fails too without that flag.

**Comment by vi4m** [comment]
@szok is it ready to merge?

File: src/ralph/data_importer/widgets.py [hunk 3]
```diff
@@ -217,3 +217,29 @@ def render(self, value):
...
+    """Widget for Rack Foreign Key field.
+
+    CSV field format Rack.name
+    """
+    def clean(self, value):
+        if not value:
+            return None
+        else:
+            try:
+                racks = Rack.objects.all()
```
Comment (szok) [inline src/ralph/data_importer/widgets.py:232]: Clean method is run for each records in CSV file. 
You can do this more performance optimal.

Pleas add some test for:
1. Import from old ralph (which Rack ID and **str**)
2. Numbers of SQL queries. (assertNumQueries)

File: src/ralph/data_importer/widgets.py [hunk 4]
```diff
@@ -217,3 +217,29 @@ def render(self, value):
...
+                data = ({'id': rack.id, '__str__': str(rack)} for rack in racks)
+                for datum in data:
+                    if datum['__str__'] == value:
+                        value = Rack.objects.get(id=datum['id'])
+                        break
+            except (ValueError, Rack.DoesNotExist):
+                value = None
+            return value
+
+    def render(self, value):
```
Comment (szok) [inline src/ralph/data_importer/widgets.py:242]: Render method must be return ID for export from GUI. Please test this
1. Import/export from GUI
2. Import/export from CLI

**Comment by szok** [comment]
@vi4m  Unfortunately it is not ready to merge. I have added new comments.

**Pull request closed without merging**